bundles/org.openhab.binding.ondilo/src/main/java/org/openhab/binding/ondilo/internal/OndiloHandler.java

```diff
@@ -0,0 +1,270 @@
+                break;
+            case "ph":
+                updateState(CHANNEL_PH, new DecimalType(lastMeasures.value));
+                break;
+            case "orp":
+                updateState(CHANNEL_ORP, new QuantityType<>(lastMeasures.value / 1000.0, Units.VOLT));
+                // Convert mV to V
+                break;
+            case "salt":
+                updateState(CHANNEL_SALT,
+                        new QuantityType<>(lastMeasures.value * 0.001, Units.KILOGRAM_PER_CUBICMETRE));
+                // Convert mg/l to kg/m³
+                break;
+            case "tds":
+                updateState(CHANNEL_TDS, new QuantityType<>(lastMeasures.value, Units.PARTS_PER_MILLION));
+                break;
+            case "battery":
+                updateState(CHANNEL_BATTERY, new QuantityType<>(lastMeasures.value, Units.PERCENT));
+                break;
+            case "rssi":
+                updateState(CHANNEL_RSSI, new DecimalType(lastMeasures.value));
+                break;
+            default:
+                logger.warn("Unknown data type: {}", lastMeasures.data_type);
+        }
+        // Update value time channel (expect that it is the same for all measures)
+        updateState(CHANNEL_VALUE_TIME, new DateTimeType(convertUtcToSystemTimeZone(lastMeasures.value_time)));
+    }
+
+    private void updateRecommendationChannels(Recommendation recommendation) {
+        updateState(CHANNEL_RECOMMENDATION_ID, new DecimalType(recommendation.id));
+        updateState(CHANNEL_RECOMMENDATION_TITLE, new StringType(recommendation.title));
+        updateState(CHANNEL_RECOMMENDATION_MESSAGE, new StringType(recommendation.message));
+        updateState(CHANNEL_RECOMMENDATION_CREATED_AT, new DateTimeType(recommendation.created_at));
+        updateState(CHANNEL_RECOMMENDATION_UPDATED_AT, new DateTimeType(recommendation.updated_at));
+        updateState(CHANNEL_RECOMMENDATION_STATUS, new StringType(recommendation.status));
+        updateState(CHANNEL_RECOMMENDATION_DEADLINE, new DateTimeType(recommendation.deadline));
+    }
+
+    private void startOndiloPolling(Integer refreshInterval) {
+        ScheduledFuture<?> currentPollingJob = ondiloPollingJob;
+        if (currentPollingJob == null) {
+            ondiloPollingJob = scheduler.scheduleWithFixedDelay(() -> poll(), 1, refreshInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    private void stopOndiloPolling() {
+        ScheduledFuture<?> currentPollingJob = ondiloPollingJob;
+        if (currentPollingJob != null) {
+            currentPollingJob.cancel(true);
+            ondiloPollingJob = null;
+        }
+    }
+
+    @Nullable
+    private OndiloBridge getOndiloBridge() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            ThingHandler handler = bridge.getHandler();
+            if (handler instanceof OndiloBridgeHandler bridgeHandler) {
```

Comment:
Think this should work
```suggestion
        if (getBridge() instanceof Bridge bridge && bridge.getHandler() instanceof OndiloBridgeHandler bridgeHandler) {
```

Reply:
sure, if readability is not important ;-)